src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs

```diff
@@ -167,6 +167,46 @@ await foreach (var response in listInstanceTypesPaginator.Responses)
             });
         }
 
+        public async Task<InstanceTypeInfo?> DescribeInstanceType(string instanceType)
+        {
+            var ec2Client = _awsClientFactory.GetAWSClient<IAmazonEC2>();
+
+            var request = new DescribeInstanceTypesRequest
+            {
+                InstanceTypes = new List<string> { instanceType }
+            };
+
+            return await HandleException(async () =>
+            {
+                try
+                {
+                    var response = await ec2Client.DescribeInstanceTypesAsync(request);
+
+                    if (response.InstanceTypes?.Count >= 1)
+                    {
+                        return response.InstanceTypes[0];
+                    }
+                    else
+                    {
+                        return null;
+                    }
+                    
+                }
+                catch (AmazonEC2Exception ex)
+                {
+                    // Check for the expected exception if the provided instance type is invalid
+                    if (string.Equals(ex.ErrorCode, "InvalidInstanceType", StringComparison.InvariantCultureIgnoreCase))
+                    {
+                        return null;
+                    }
+                    else
+                    {   // Anything else is unexpected, so proceed with usual exception handling
+                        throw ex;
+                    }
+                }
```

Comment:
this logic should be added in the validator and not here. We shouldn't add business logic in the AWSResourceQueryer.  Let the service call error out if the instance type isn't found and catch that error in the validator.

Reply:
Why do we not want to add business logic in AWSResourceQueryer? There's no documentation on the interface. Could you point me to a design doc with intent?

----

We have similar logic around resource existence already in:
1. GetECRRepositories - https://github.com/aws/aws-dotnet-deploy/blob/e2ad9b5a026869aa8854182dd5df1f4cedc6fa46/src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs#L533-L536
2. GetCloudFormationStack - https://github.com/aws/aws-dotnet-deploy/blob/e2ad9b5a026869aa8854182dd5df1f4cedc6fa46/src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs#L573-L577
3. DescribeECRRepository - https://github.com/aws/aws-dotnet-deploy/blob/e2ad9b5a026869aa8854182dd5df1f4cedc6fa46/src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs#L711-L714
4. ParameterNotFoundException - https://github.com/aws/aws-dotnet-deploy/blob/e2ad9b5a026869aa8854182dd5df1f4cedc6fa46/src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs#L731-L734

Should I rework all of those too?

Reply:
We don't have that documented anywhere, but since AWSResourceQueryer is basically doing some SDK calls to retrieve resources, I'd like to leave them as raw as possible to promote reusability of those methods. Any needed business logic can be added in the caller method. As for the other examples where we are not doing that, I usually update the methods whenever I am working on a change that touches those methods. So I wouldn't update the rest in this PR to avoid the scope creep.

Reply:
Moved the exception handling to the validator, and added a note on the interface for the future.